core/sequencer/server.go

```diff
@@ -244,8 +244,8 @@ func (s *Server) readMessages(ctx context.Context, directoryID string, meta *spb
 	batchSize int32) ([]*mutator.LogMessage, error) {
 	msgs := make([]*mutator.LogMessage, 0)
 	for _, source := range meta.Sources {
-		low := source.GetLowestWatermark()
-		high := source.GetHighestWatermark()
+		low := source.GetLowestInclusive()
```

Comment:
Can `source` be nil? Maybe just `source.HighestWatermark`, like in other similar lines below.